lib/src/model/comment_referable.dart

```diff
@@ -97,27 +97,40 @@ mixin CommentReferable implements Nameable {
     _ReferenceChildrenLookup referenceLookup, {
     required bool Function(CommentReferable?) filter,
   }) {
-    final resultElement = scope!.lookupPreferGetter(referenceLookup.lookup);
-    if (resultElement == null) return null;
+    Element? resultElement;
+    final scope = this.scope;
+    if (scope != null) {
+      resultElement = scope.lookupPreferGetter(referenceLookup.lookup);
+      if (resultElement == null) return null;
+    }
+
+    if (this case ModelElement(:var modelNode?) when resultElement == null) {
```

Comment:
Cleaver, maybe too much :-P
Match a pattern, and then check a condition that does not use the pattern.

Ideally, should be a separate `if` enclosing the pattern matching one, but one liner is tempting :-)

Reply:
Yeah I definitely thought this was a little weird :P Specifically because it seems like the `when resultElement == null` should be first. But we cannot write `if (resultElement == null && this case ModelElement(:var modelNode?))`. I can split it up in a follow-up.